sdk/src/main/scala/io/horizen/account/state/ForgerStakeMsgProcessor.scala

```diff
@@ -445,29 +446,57 @@ case class ForgerStakeMsgProcessor(params: NetworkParams) extends NativeSmartCon
     restrictForgerList
   }
 
-  @throws(classOf[ExecutionFailedException])
-  override def process(invocation: Invocation, view: BaseAccountStateView, context: ExecutionContext): Array[Byte] = {
-    val gasView = view.getGasTrackedView(invocation.gasPool)
-    getFunctionSignature(invocation.input) match {
-      case GetPagedListOfForgersCmd if Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active
-                               => doGetPagedListOfForgersCmd(invocation, gasView)
-      case GetListOfForgersCmd => doGetListOfForgersCmd(invocation, gasView, Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active)
-      case AddNewStakeCmd => doAddNewStakeCmd(invocation, gasView, context.msg,
-                              Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active)
-      case RemoveStakeCmd => doRemoveStakeCmd(invocation, gasView, context.msg, Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active)
-      case OpenStakeForgerListCmd => doOpenStakeForgerListCmd(invocation, gasView, context.msg)
-      case OpenStakeForgerListCmdCorrect if Version1_2_0Fork.get(context.blockContext.consensusEpochNumber).active
-                                => doOpenStakeForgerListCmd(invocation, gasView, context.msg)
-      case UpgradeCmd if Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active
-                                => doUpgradeCmd(invocation, view)// This doesn't consume gas, so it doesn't use GasTrackedView
-      case StakeOfCmd if Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active
-                                => doStakeOfCmd(invocation, gasView)
-      case GetPagedForgersStakesOfUserCmd if Version1_3_0Fork.get(context.blockContext.consensusEpochNumber).active
-                                => doGetPagedForgersStakesOfUserCmd(invocation, gasView)
-      case opCodeHex => throw new ExecutionRevertedException(s"op code not supported: $opCodeHex")
-    }
+  def doDisable(invocation: Invocation, view: BaseAccountStateView): Array[Byte] = {
+    requireIsNotPayable(invocation)
+    checkInputDoesntContainParams(invocation.input)
+    if (WellKnownAddresses.FORGER_STAKE_V2_SMART_CONTRACT_ADDRESS != invocation.caller ||
+      !view.getCodeHash(invocation.caller).sameElements(ForgerStakeV2MsgProcessor.contractCodeHash))
+      throw new ExecutionRevertedException("Authorization failed")
+
+    ForgerStakeStorage.setDisabled(view)
```

Comment:
As for `setActive` before, I don't see a lower risk calling directly `view.updateAccountStorage ` instead of `ForgerStakeStorage.setDisabled(view)` and I prefer keeping the info about its key in the same place.
As for the addLog, I have moved the one related to V2 instead.